cerebrium/container-images/defining-container-images.mdx

```diff
@@ -0,0 +1,355 @@
+
+### Conda Packages
+
+Conda excels at managing complex system-level Python dependencies, particularly for GPU support and scientific computing:
+
+```toml
+[cerebrium.dependencies.conda]
+cuda = ">=11.7"
+cudatoolkit = "11.7"
+opencv = "latest"
+```
+
+Teams using conda environments can specify their environment file:
+
+```toml
+[cerebrium.dependencies.paths]
+conda = "conda_pkglist.txt"
+```
+
+Like APT packages, Conda packages often modify system-level components. Changes to Conda dependencies will trigger a full rebuild to ensure all binary dependencies and system libraries are correctly configured. Consider batching Conda dependency updates together to minimize rebuild time.
+
+## Setup tasks
+
+Cerebrium provides two distinct types of commands for handling setup tasks. Understanding the difference helps optimize build times and ensures reliable app initialization.
+
+### Shell Commands
+
+Shell commands execute every time a new instance starts, making them ideal for runtime initialization and app startup:
+
+```toml
+shell_commands =  [
+    "echo 'Hello, World!'",
+    "curl -LsSf https://astral.sh/uv/install.sh | sh",
+    "uv venv",
+    "uv pip install transformers",
+    "export SOME_ENV_VAR=value"
+]
+```
+
+Shell commands are perfect for tasks that need fresh execution on startup, such as downloading models, warming caches, or starting an app server. Since these commands run on every instance start, they should be optimized for speed to minimize startup time.
+
+### Pre-build Commands
+
+Pre-build commands execute once during image creation, making them suitable for setup tasks that can be baked into the container (Which should also improve cold start performance):
+
+```toml
+pre_build_commands = [
+    "curl -o /usr/local/bin/pget -L 'https://github.com/replicate/pget/releases/download/v0.6.2/pget_linux_x86_64' && chmod +x /usr/local/bin/pget"
+]
+```
+
+Pre-build commands are ideal for compiling code, building assets, or installing build-time dependencies. Since these commands affect the container image, changes to pre-build commands trigger a full rebuild. Group related commands together to minimize rebuild frequency.
+
+## Custom Docker Base Images
+
+The choice of base image fundamentally shapes how an app runs in Cerebrium. While the default Debian slim image works well for most Python apps, some scenarios require specialized base images.
+
+### Selecting Base Images
+
+The base image is specified through the `docker_base_image_url` parameter:
+
+```toml
+# Default minimal image
+docker_base_image_url = "debian:bookworm-slim"
+
+# CUDA-enabled image for GPU workloads
+docker_base_image_url = "nvidia/cuda:12.0.1-runtime-ubuntu22.04"
+docker_base_image_url = "nvidia/cuda:12.0.1-devel-ubuntu22.04"
+```
+
+<Info>
+  Cerebrium supports all public docker images that are compatible with Ubuntu,
+  Cuda, and Python
+</Info>
+
```

Comment:
this is not correct, we support the following repos: `nvidia/cuda` with ubuntu, `debian:*`, `ubuntu:*` and `python` with ubuntu